src/osx/setup-env.cpp

```diff
@@ -28,14 +58,22 @@ void setupEnvironment() {
     auto imModuleFile = libPath / "gtk-3.0" / "3.0.0" / "immodules.cache";
     auto pixbufModuleFile = libPath / "gdk-pixbuf-2.0" / "2.10.0" / "loaders.cache";
 
-    setenv("XDG_CONFIG_DIRS", xdgPath.string().c_str(), 1);
-    setenv("XDG_DATA_DIRS", dataPath.string().c_str(), 1);
-    setenv("GTK_DATA_PREFIX", base.string().c_str(), 1);
-    setenv("GTK_EXE_PREFIX", base.string().c_str(), 1);
-    setenv("GTK_PATH", base.string().c_str(), 1);
+    // Override any environment variables corresponding to install prefix paths.
+    // Only do this if the paths don't exist, otherwise we could run into issues
+    // (e.g., nonexistent pixbuf module file will lead to a startup crash)
+    setenvPathIfExists("XDG_CONFIG_DIRS", xdgPath, 1);
+    setenvPathIfExists("XDG_DATA_DIRS", dataPath, 1);
```

Comment:
The behavior of overriding (instead of prepending) is too aggressive in the use case I am interested in (xournalpp with Nix), since `XDG_DATA_DIRS` may contain the directories with the system themes (e.g., Adwaita and hicolor theme).

For example, xournalpp logs the following on startup:
```
Skip override env var XDG_CONFIG_DIRS, path doesn't exist: "/nix/store/j4b0xb53icgpa60pybfvdpl8kd1w6mlq-xournalpp-1.1.1/etc/xdg"
Overriding env var XDG_DATA_DIRS: "/nix/store/j4b0xb53icgpa60pybfvdpl8kd1w6mlq-xournalpp-1.1.1/share"
```

However, my `XDG_DATA_DIRS` already consists of:
```
/nix/store/j4b0xb53icgpa60pybfvdpl8kd1w6mlq-xournalpp-1.1.1/share:/nix/store/iiiib5v00l61fbf7fp12hnd2nfkp2kf9-hicolor-icon-theme-0.17/share:/nix/store/f0j672ahrxq9ag0ldq3w771l40fr90ls-adwaita-icon-theme-42.0/share:
```

So for these "DIRS" variables, it would be better to prepend the paths. The downside is that the user may have potentially already set one of these variables, which can cause confusion.